### PR TITLE
fix(apps): validate if the app exists and is published before installation

### PR DESCRIPTION
**Pull request by oioki**
* fixes [HTTP 500 on SentryApp.DoesNotExist](https://sentry.sentry.io/issues/4914934320/?project=1), now responds with empty 404
* disallows installation of other org's unpublished sentry apps
* disallows installation of other org's internal integration as a sentry app

### CODE REVIEW

File: src/sentry/api/endpoints/integrations/sentry_apps/installation/index.py [hunk 1]
```diff
@@ -43,9 +44,17 @@ def post(self, request: Request, organization) -> Response:
...
             return Response(serializer.errors, status=400)
 
-        # check for an exiting installation and return that if it exists
         slug = serializer.validated_data.get("slug")
+
+        # only published or owned apps are allowed to be installed
+        app = SentryApp.objects.filter(slug=slug).first()
+        if not app:
+            return Response(status=404)
+        if not app.status == SentryAppStatus.PUBLISHED and app.owner_id != organization.id:
```
Comment (wedamija) [inline src/sentry/api/endpoints/integrations/sentry_apps/installation/index.py:53]: Should this be an OR?

Comment (wedamija) [inline src/sentry/api/endpoints/integrations/sentry_apps/installation/index.py:53]: Nit: May as well combine this all into one statement?

Comment (oioki) [reply src/sentry/api/endpoints/integrations/sentry_apps/installation/index.py]: Here is the truth table. We allow the app setup if it is published or if it belongs to the same org. Negating, we disallow only if it is not published AND does not belong to the same org.
```
|                                         | app.owner_id == organization.id | app.owner_id != organization.id |
|-----------------------------------------|---------------------------------|---------------------------------|
| app.status == SentryAppStatus.PUBLISHED | False (pass)                    | False (pass)                    |
| app.status != SentryAppStatus.PUBLISHED | False (pass)                    | True (return 404)               |
```

Comment (wedamija) [reply src/sentry/api/endpoints/integrations/sentry_apps/installation/index.py]: Right, that makes sense!

**Review by wedamija — APPROVED** [review]

**Pull request merged**